### DISCUSSION TITLE
Allure report Test body displaying direct click element as argument. Unable to add descriptive steps

### DISCUSSION THREAD

**Discussion by nkr741** [allure-framework/allure2#2813] [Questions & Support]
"Why are arguments being displayed at the element level in the Allure report, instead of being grouped with the corresponding step? Is there a way to configure Allure to display arguments at the step level, providing a more concise and readable report?" I am trying to add multiple ways to get the steps in the report however it is giving me allure.step() is not a function. Can we discuss and resolve the issue so that it will be a great pleasure if it is resolved.

**Comment by baev** [comment]
Could you please provide more details on the issue? A code sample and screenshot will help a lot.

**Reply by nkr741** [reply]
![image](https://github.com/user-attachments/assets/a0e457a5-642e-443c-bd03-304bb9a73dbd)
this is the cypress.config.js file
const { defineConfig } = require('cypress');
const fs = require('fs-extra');
const path = require('path');
const {allureCypress} = require('allure-cypress/reporter')
const os = require('os')
module.exports = defineConfig({
   e2e: {
       specPattern: 'cypress/e2e/**/*.cy.{js,jsx,ts,tsx}',
       baseUrl: '[https://dev.abdgh.com/login']
       viewportWidth: 1920,
       viewportHeight: 1080,
       defaultCommandTimeout: 20000,
       setupNodeEvents(on, config) {
           const reportsPath = 'cypress/reports';
           const accessibilityPath = `${reportsPath}/accessibility`;
           const functionalPath = `${reportsPath}/functional`;
           const allurePath = `${functionalPath}/allure-results`;
           const downloadsPath = 'cypress/downloads';
           [reportsPath, accessibilityPath, functionalPath, allurePath, downloadsPath].forEach(dir => {
               fs.ensureDirSync(dir);
           });
           allureCypress(on, config, {
               resultsDir: allurePath,
               reportPageTitle: 'Vermont CFIS Test Report',
               environmentInfo: {
                   os_platform: os.platform(),
                   os_version: os.version(),
                   node_version: process.version
               },
               stepLevel: true,
               stepTimings: true,
               disableWebdriverStepsReporting: true,
               addConsoleLogs: false,
               addScreenshots: true
           });
           on('task', {
               log(message) {
                   console.log(message);
                   return null;
               },
               saveA11yReport({ reportData, suiteName }) {
                   const sanitizedSuiteName = suiteName.replace(/[^a-z0-9]/gi, '-').toLowerCase();
                   const reportPath = path.join(accessibilityPath, `${sanitizedSuiteName}-accessibility-report.html`);
                   fs.writeFileSync(reportPath, reportData);
                   return null;
               },
               getTimestamp() {
                   return Date.now();
               },
               deleteDownloadedFile(filename) {
                   const downloadedFilePath = path.join('cypress', 'downloads', filename);
                   try {
                       if (fs.existsSync(downloadedFilePath)) {
                           fs.removeSync(downloadedFilePath);
                           return true;
                       }
                       return null;
                   } catch (err) {
                       return `Failed to delete file: ${err.message}`;
                   }
               },
               verifyDownload(filename) {
                   const downloadedFilePath = path.join('cypress', 'downloads', filename);
                   return fs.existsSync(downloadedFilePath);
               }
           });
           return config;
       }
   },
   watchForFileChanges: false,
   chromeWebSecurity: false,
   video: false,
   screenshotOnRunFailure: true,
   screenshotsFolder: 'cypress/reports/screenshots',
   videosFolder: 'cypress/reports/videos',
   retries: {
       runMode: 1,
       openMode: 0
   },
   downloadsFolder: 'cypress/downloads',
   defaultCommandTimeout: 20000,
   execTimeout: 20000,
   taskTimeout: 20000,
   pageLoadTimeout: 20000,
   requestTimeout: 20000,
   responseTimeout: 20000
})

if you see the screenshot the test steps are not as exxpected. When i use allure.step() it is throwing error as 
Cannot read properties of undefined (reading 'step')
Page Objects/Page Actions/Common Actions.js:11:11
   9 | */
  10 | AdminLogin(){
> 11 | cy.allure.step('Login as Admin', () =>{
     |           ^
  12 |     cy.get("mat-tab-body div div div a div mat-icon[role='img']").click();
  13 |     cy.get(CommonUtils.userName).type(Cypress.env('userName'))
  14 |     cy.get(CommonUtils.password).type(Cypress.env('password')

**Reply by nkr741** [reply]
I have followed the same configuration steps given in the documentation yet getting this issue. I am using latest cypress version and this is my package.json file for the dependencies i am using.
  "scripts": {
    "clean:reports": "rimraf \"cypress/reports/functional/allure-results\" \"allure-results\" \"allure-report\"",
    "cypress:run": "cypress run --browser chrome",
    "generate:report": "allure generate cypress/reports/functional/allure-results --clean -o allure-report --single-file",
    "serve:report": "allure serve cypress/reports/functional/allure-results",
    "test:email-report": "npm run clean:reports && npm run cypress:run && npm run generate:report",
    "test:view-report": "npm run clean:reports && npm run cypress:run && npm run serve:report"
  },
  "author": "Naveen Kumar Manda",
  "license": "ISC",
  "description": "",
  "devDependencies": {
    "@axe-core/watcher": "^3.14.3",
    "@cypress/code-coverage": "^3.12.48",
    "@faker-js/faker": "^9.0.0",
    "allure-commandline": "^2.30.0",
    "allure-cypress": "^3.0.6",
    "allure-js-commons": "^3.0.6",
    "axe-core": "^4.8.2",
    "axe-reporter-html": "^2.0.0",
    "axe-reports": "^1.1.11",
    "cypress": "^13.16.0",
    "cypress-ag-grid": "^3.3.1",
    "cypress-axe": "^1.5.0",
    "cypress-downloadfile": "^1.2.4",
    "cypress-email-results": "^1.8.0",
    "cypress-file-upload": "^5.0.8",
    "cypress-multi-reporters": "^1.5.0",
    "cypress-parallel": "^0.14.0",
    "cypress-plugin-api": "^2.11.2",
    "cypress-wait-until": "^3.0.2",
    "cypress-xpath": "^2.0.1",
    "dotenv": "^16.4.5",
    "exceljs": "^4.4.0",
    "glob": "^11.0.0",
    "log4js": "^6.9.1",
    "mocha": "^10.8.2",
    "mocha-junit-reporter": "^2.2.1",
    "mysql2": "^3.11.2",
    "nodemailer": "^6.9.15",
    "pdf-parse": "^1.1.1",
    "pg": "^8.12.0",
    "rimraf": "^6.0.1",
    "xlsx": "^0.18.5"
  }
}

**Comment by baev** [comment]
Instead of calling `cy.allure.step` you should import `step` function from `allure-js-commons`:

```js
import * as allure from "allure-js-commons";
import { Status } from "allure-js-commons";

it("Authentication", () => {
  allure.step("Step 1", () => {
    allure.step("Sub-step 1", () => {
   
    });
    allure.step("Sub-step 2", () => {
    });
  });
  allure.logStep("Step 2", Status.SKIPPED);
});
```

For more details, please see the docs https://allurereport.org/docs/cypress/#divide-a-test-into-steps

**Reply by nkr741** [reply]
Thank you @baev for the solution. I am half way to my expectation. I am able to get the steps but still i am getting the component level sub steps. i tried the config to make if false even i am getting the element displaying in the report. I wany only the High level Action steps to display not the sub steps. Can you please help me with that.
![image](https://github.com/user-attachments/assets/e155170b-c53d-4442-a7c8-1699a93395ba)
![image](https://github.com/user-attachments/assets/64e5b4f8-8ea9-4b90-bc74-fc315448b6a0)
.

**Reply by nkr741** [reply]
i made them false
![image](https://github.com/user-attachments/assets/5a19eea9-a2e2-4368-8026-67f3668d6aea)
